lib/statesman/adapters/active_record.rb

```diff
@@ -70,7 +70,7 @@ def create_transition(from, to, metadata)
 
         transition = transitions_for_parent.build(transition_attributes)
 
-        ::ActiveRecord::Base.transaction do
+        @parent_model.with_lock do
```

Comment:
Is there a reason to lock here (around the whole transaction) rather than just around `unset_most_recent`?